src/test/java/app/component/Operator.java

```diff
@@ -182,6 +201,125 @@ public JsonNode v2TokenGenerateUsingPayload(String payload, boolean asOldPartici
         return v2DecryptEncryptedResponse(encryptedResponse, envelope.nonce(), getClientApiSecret(asOldParticipant));
     }
 
+    public JsonNode v2ClientSideTokenGenerate(String requestBody, boolean useValidOrigin) throws Exception {
+        final byte[] serverPublicKeyBytes = base64ToByteArray(CLIENT_SIDE_TOKEN_GENERATE_SERVER_PUBLIC_KEY.substring(PUBLIC_KEY_PREFIX_LENGTH));
+
+        final PublicKey serverPublicKey = KeyFactory.getInstance("EC")
+                .generatePublic(new X509EncodedKeySpec(serverPublicKeyBytes));
+
+        final KeyPair keyPair = generateKeyPair();
+        final SecretKey sharedSecret = generateSharedSecret(serverPublicKey, keyPair);
+
+        final JsonObject cstgEnvelope = createCstgEnvelope(requestBody, CLIENT_SIDE_TOKEN_GENERATE_SUBSCRIPTION_ID, keyPair.getPublic(), sharedSecret);
+
+        final Request.Builder requestBuilder = new Request.Builder()
+                .url(getBaseUrl() + "/v2/token/client-generate")
+                .addHeader("Origin", useValidOrigin ? CLIENT_SIDE_TOKEN_GENERATE_ORIGIN : CLIENT_SIDE_TOKEN_GENERATE_INVALID_ORIGIN)
+                .post(RequestBody.create(cstgEnvelope.toString(), HttpClient.JSON));
+
+        final String encryptedResponse = HttpClient.execute(requestBuilder.build(), HttpClient.HttpMethod.POST);
+        final byte[] decryptedResponse = decrypt(base64ToByteArray(encryptedResponse), sharedSecret);
+
+        return Mapper.OBJECT_MAPPER.readTree(decryptedResponse);
+    }
+
+    private static KeyPair generateKeyPair() {
+        final KeyPairGenerator keyPairGenerator;
+        try {
+            keyPairGenerator = KeyPairGenerator.getInstance("EC");
+        } catch (NoSuchAlgorithmException e) {
+            throw new RuntimeException(e);
+        }
+        final ECGenParameterSpec ecParameterSpec = new ECGenParameterSpec("secp256r1");
+        try {
+            keyPairGenerator.initialize(ecParameterSpec);
+        } catch (InvalidAlgorithmParameterException e) {
+            throw new RuntimeException(e);
+        }
+        return keyPairGenerator.genKeyPair();
+    }
+
```

Comment:
As @gmsdelmundo mentioned, any way to share this code with Operator (test) code?